### DISCUSSION TITLE
Deep Link for activation

### DISCUSSION THREAD

**Discussion by elKei24** [digitalfabrik/entitlementcard#179] [General]
Moved from discord. Summary:
* Easy to use App Links / Universial in order to if (s)he accidentally scans the card activation QR code using her/his usual QR scanner app
  * and app is not installed then lead her/him to our website (and then to playstore) 
  * and app is installed then activate the card
* Problem: card data may be stored in browser history

**Comment by elKei24** [comment]
@maxammann I changed the nginx config to rewrite the request so that the "card" parameter is discarded. It is not visible in my browser history now any more. Still not sure if this solves the problem or just "hides" it for me in the UI but still exposes the card data. Do you know that?

**Reply by maxammann** [reply]
rewrite redirects the client as far as I could find out. There are no specifications which say that browsers or browser extensions can not or should not log the initial request. That means just because firefox is doing this does not mean that the Samsung Browser also does this.

It still appears in the log which could be fixed of course by disabling logging. But this is still dangerous terrain:

```
**** - - [15/Jan/2021:10:05:40 +0100] "GET /?card=Ibims1testVongEliasHer HTTP/1.1" 301 178 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:84.0) Gecko/20100101 Firefox/84.0"
```

I think there is no way to fix this sadly in an absolute secure manner. Having secrets in the URL of a GET request is dangerous and should only be done for unimportant stuff.

**Reply by elKei24** [reply]
Hm, sad. We probably have to move away from the idea of using deep links completely then. Using a custom URL schema is also insecure as anyone can register a handler for that and so grab our data.

**Reply by maxammann** [reply]
oh oke really? that sucks a lot. But again we have to think about convenience. I mean if we can check that no browser keeps a history of redirects than we can go with the first option.

If any app can register deep links then also the first version with "https" and rewrites is "insecure". But I think if you have malicous apps on your phone then your are already fucked.

**Reply by elKei24** [reply]
Not really, that is the advantage of the https-Links, because there Android/iOS checks if we as the domain owners allow  a specific app to handle URLs that belong to our domain. I had to put two files in the .well-known folder of our web server containing e.g. the signing key signature.

Another possibility would be to manipulate the QR-Code, so that "usual" readers ignore the card data and just see an URL but our reader can still extract the card data. No idea if something like that is possible.

**Reply by maxammann** [reply]
I also though about that, but did not find anything quickly on google

**Reply by elKei24** [reply]
Me neither.

I also tried it using `https://user:carddata@activate.ehrenamtdkarte.app`, but even if Firefox warned me about unrequested login, it then stored the login data in the browser history 🙁
